digdag-core/src/test/java/io/digdag/core/workflow/WorkflowExecutorTest.java

```diff
@@ -434,6 +438,61 @@ public void randomEnqueue()
         }
     }
 
+    @Test
+    public void verifySubmittedTasks()
+            throws Exception
+    {
+        Instant startTime = Instant.now().truncatedTo(ChronoUnit.SECONDS);
+        StoredSessionAttemptWithSession attempt = runWorkflow("nested", loadYamlResource("/io/digdag/core/workflow/nested.dig"));
+        embed.getLocalSite().runUntilDone(attempt.getId());
+        Instant finishTime = Instant.now().plusSeconds(1).truncatedTo(ChronoUnit.SECONDS);
+
+        List<ArchivedTask> tasks = tm.begin(() -> embed.getLocalSite().getSessionStore().getTasksOfAttempt(attempt.getId()));
+        assertEquals(3, tasks.size());
+
+        Instant rootTaskStartedAt;
+        {
+            // Root task
+            StoredTask task = tasks.get(0);
+            assertEquals("+nested", task.getFullName());
+            assertTrue(task.getTaskType().isGroupingOnly());
+            assertEquals(TaskStateCode.SUCCESS, task.getState());
+            assertThat(task.getUpdatedAt(), is(greaterThanOrEqualTo(startTime)));
+            assertThat(task.getUpdatedAt(), is(lessThanOrEqualTo(finishTime)));
+            assertThat(task.getStartedAt().get(), is(greaterThanOrEqualTo(startTime)));
```

Comment:
[IMO] IIUC, the purpose of this assertion is to confirm that a group taks' `started_at` was updated (sorry if I mis-understood about it). So I think it would be better to leave the comment to describe the purpose for better clarification.